### ISSUE TITLE
[BUG] It is possible to generate a Organization with no name.

### ISSUE THREAD

**Issue by Xniveres** [appsmithorg/appsmith#774]
## Description

[What happened]
I was able to generate a Organization with no name with a manually generated HTTP request.

### Steps to reproduce the behaviour:
Send a HTTP request like this:

```
POST /api/v1/organizations HTTP/1.1
Host: app.appsmith.com
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:80.0) Gecko/20100101 Firefox/80.0
Accept: application/json, text/plain, */*
Accept-Language: de,en-US;q=0.7,en;q=0.3
Accept-Encoding: gzip, deflate
Content-Type: application/json
Content-Length: 18
Origin: https://app.appsmith.com
Referer: https://app.appsmith.com/applications
Connection: close
Cookie: SESSION=notstealingmycookie

{"name":""}
```

### Important Details
 - Version: Cloud
 - OS: Windows
 - Browser Chrome

**Comment by Nikhil-Nandagopal** [comment]
@Xniveres nice carch!

**Comment by avats-dev** [comment]
@Nikhil-Nandagopal I can also work on this one if anyone can help me a bit in resolving this. Thanks.

**Comment by gilga98** [comment]
can I pick this up?

**Comment by Nikhil-Nandagopal** [comment]
@gilga98 I have to give this to @avats-dev since he claimed it first! But we have a lot of good first issues that you can pick up.
@trishaanand can you guide us on the solution here

**Comment by trishaanand** [comment]
In the file OrganizationServiceImpl.java, you will find the function : `public Mono<Organization> create(Organization organization, User user)`
This function is hit while creating organization. Inside this function, we should check for the organization.getName() being non-empty. In case that's not true, we should throw the error : `AppsmithError.INVALID_PARAMETER`

**Comment by avats-dev** [comment]
@trishaanand @Nikhil-Nandagopal thanks. I will get started on this one. :+1:

**Comment by Nikhil-Nandagopal** [comment]
@avats-dev I don't see a PR for this yet so I'm unassigning you. Feel free to come back to this when you have the time :)

**Comment by mohanarpit** [comment]
@all-contributors Please add @Xniveres for their invaluable contribution towards bug reporting.

**Comment by daniloff200** [comment]
Looks like nobody pick that for now.
@mohanarpit  can I pick this one ?

**Comment by daniloff200** [comment]
Should be closed by #1120, not sure, why it didn't
